### PR TITLE
[Fix] Add ignore_index parameter in L1 loss in release/2.6

### PR DESCRIPTION
**Pull request by shiyutang**
fix issue https://github.com/PaddlePaddle/PaddleSeg/issues/2354
验证有无ignoreindex在三种reduction下均得到相同结果，且mean方式下，backward正常传播。

image
image
image

### CODE REVIEW

File: paddleseg/models/losses/l1_loss.py [hunk 1]
```diff
@@ -74,3 +73,14 @@ class L1Loss(nn.L1Loss):
...
+        self.ignore_index = ignore_index
+    
+    def forward(self, input, label):
+        mask = label==self.ignore_index
+        mask.stop_gradient = True
+        label[mask] = 0
+        input[mask] = 0
+        
+        return paddle.nn.functional.l1_loss(
+            input, label, self.reduction, name=self.name)
```
Comment (wuyefeilin) [inline paddleseg/models/losses/l1_loss.py]: 返回数值应该是有效像素的平均值

File: paddleseg/models/losses/l1_loss.py [hunk 2]
```diff
@@ -74,3 +73,14 @@ class L1Loss(nn.L1Loss):
 
     def __init__(self, reduction='mean', ignore_index=255):
         super().__init__(reduction=reduction)
+        self.ignore_index = ignore_index
+    
+    def forward(self, input, label):
+        mask = label==self.ignore_index
+        mask.stop_gradient = True
+        label[mask] = 0
+        input[mask] = 0
```
Comment (juncaipeng) [inline paddleseg/models/losses/l1_loss.py]: 这样直接赋值为0，后面计算loss取平均的时候，会把需要忽略的像素也算进来取平均了。可以参考cross_entropy_loss中的mask处理方式进行修改。

**Review by wuyefeilin — CHANGES_REQUESTED** [review]

File: paddleseg/models/losses/l1_loss.py (hunk 2 above)
Comment (shiyutang) [reply paddleseg/models/losses/l1_loss.py]: cross_entropy loss 是function可以传入ignore index的，和这里不一样，我重新处理下平均这个问题。

File: paddleseg/models/losses/l1_loss.py (hunk 1 above)
Comment (shiyutang) [reply paddleseg/models/losses/l1_loss.py]: 嗯嗯，好的

File: paddleseg/models/losses/l1_loss.py [hunk 3]
```diff
@@ -74,3 +73,32 @@ class L1Loss(nn.L1Loss):
...
+
+        if invalid_label_cnt == 0:
+            return paddle.nn.functional.l1_loss(
+                    input, label, self.reduction, name=self.name)
+        else:
+            output = paddle.nn.functional.l1_loss(
+                input, label, "none", name=self.name)
+        
+            if self.reduction == "mean":
+                return paddle.sum(output)/valid_label_cnt
```
Comment (wuyefeilin) [inline paddleseg/models/losses/l1_loss.py]: 防止分母为0的情况，即label都是ignore_index的情况

Comment (shiyutang) [reply paddleseg/models/losses/l1_loss.py]: done

**Review by wuyefeilin — APPROVED** [review]
LGTM

**Review by juncaipeng — APPROVED** [review]
LGTM
